doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -0,0 +1,970 @@
+    :n:`[x₀; ...; xₙ]`.
+
+- :n:`list0(@scope, sep = @string)`:
+
+  + if :n:`@scope` parses :n:`@entry`, parses `(x₀ = @entry, sep, ..., sep, xₙ = @entry)`
+    and produces `[x₀; ...; xₙ]`.
+
+- :n:`list1`: same as :n:`list0` (with or without separator) but parses :n:`{+ @entry}` instead
+  of `ENTRY*`.
+
+- :n:`opt(@scope)`
+
+  + if :n:`@scope` parses :n:`@entry`, parses :n:`{? @entry}` and produces either :n:`None` or
+    :n:`Some x` where :n:`x` is the parsed expression.
+
+- :n:`self`:
+
+  + parses a Ltac2 expression at the current level and return it as is.
+
+- :n:`next`:
+
+  + parses a Ltac2 expression at the next level and return it as is.
+
+- :n:`tactic(n = @int)`:
+
+  + parses a Ltac2 expression at the provided level :n:`n` and return it as is.
+
+- :n:`thunk(@scope)`:
+
+  + parses the same as :n:`scope`, and if :n:`e` is the parsed expression, returns
+    :n:`fun () => e`.
+
+- :n:`STRING`:
+
+  + parses the corresponding string as a CAMLP5 IDENT and returns :n:`()`.
+
+- :n:`keyword(s = @string)`:
+
+  + parses the string :n:`s` as a keyword and returns `()`.
+
+- :n:`terminal(s = @string)`:
+
+  + parses the string :n:`s` as a keyword, if it is already a
+    keyword, otherwise as an :n:`@ident`. Returns `()`.
+
+- :n:`seq(@scope__1, ..., @scope__2)`:
+
+  + parses :n:`scope__1`, ..., :n:`scope__n` in this order, and produces a tuple made
+    out of the parsed values in the same order. As an optimization, all
+    subscopes of the form :n:`STRING` are left out of the returned tuple, instead
+    of returning a useless unit value. It is forbidden for the various
+    subscopes to refer to the global entry using self of next.
+
+A few other specific scopes exist to handle Ltac1-like syntax, but their use is
+discouraged and they are thus not documented.
+
+For now there is no way to declare new scopes from Ltac2 side, but this is
+planned.
+
+Notations
+~~~~~~~~~
+
+The Ltac2 parser can be extended by syntactic notations.
+
+.. cmd:: Ltac2 Notation {+ @var (@scope) %| @string } {? : @int} := @term
+   :name: Ltac2 Notation
+
+   A Ltac2 notation adds a parsing rule to the Ltac2 grammar, which is expanded
+   to the provided body where every token from the notation is let-bound to the
+   corresponding generated expression.
+
+   For instance, assume we perform:
+
+   :n:`Ltac2 Notation "foo" c(thunk(constr)) ids(list0(ident)) := Bar.f c ids.`
+
+   Then the following expression
+
+   :n:`let y := @X in foo (nat -> nat) x $y`
+
+   will expand at parsing time to
+
+   :n:`let y := @X in`
+   :n:`let c := fun () => constr:(nat -> nat) with ids := [@x; y] in Bar.f c ids`
+
+   Beware that the order of evaluation of multiple let-bindings is not specified,
+   so that you may have to resort to thunking to ensure that side-effects are
+   performed at the right time.
+
+Abbreviations
+~~~~~~~~~~~~~
+
+.. cmdv:: Ltac2 Notation @ident := @term
```

Comment:
Are you sure this is `ident` and not `lident`. And what about `term`? Should it be `ltac2_term`? Same question for the command defined above and for the command defined below.